src/mainwindow.cpp

```diff
@@ -111,16 +119,14 @@ MainWindow::MainWindow(const QString& work_dir,
 
 void MainWindow::rebuildActions()
 {
-    QMap< QString, QAction * > oldActions(actions);
+    // Delete all setting-related QObjects
+    delete settingOwner;
+    settingOwner = new QWidget(this);
```

Comment:
If menus are to be re-parented, then yes, QWidget is necessary. But since menus are not re-parented due to DBusMenuExporterDBus warnings, it's not that necessary. Can be useful in future for other setting-related QWidgets, so I'd say it's justified.